### PR TITLE
feat(CART-1044): See users' carts with a dynamic user interface

### PR DESCRIPTION
**Pull request by AlexCristea04**
**JIRA:** link to jira ticket
## Context:
This ticket adds react pages for the user carts and the list of all carts. Additionally, I have added API gateway implementation, as our endpoints were not working before, therefore I could not tie the frontend to the backend. Get all has been implemented, and all carts are displayed using it.

## Does this PR change the .vscode folder in petclinic-frontend?:
No.

## Changes
- Added API gateway implementation (service client, controller, DTOs)
- Added UserCart, CartTable, CartItem react components, along with their individual CSS files
- Fetched the carts within CartTable using get all 
- Created the cart page, modified the nav bar
- Added routing 

## Before and After UI (Required for UI-impacting PRs)
{A33BC9E9-DDEB-4EC1-8C68-40789E5A8B70}
{D202EB85-B27F-4224-B26A-C722E4D3CE7E}

### CODE REVIEW

File: petclinic-frontend/src/features/carts/components/CartItem.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                </div>
+            </div>
+            <div className="CartItem-price">
+                {formatPrice(productSalePrice)}
+            </div>
+        </div>
+    );
+};
+
+export default CartItem;
```
Comment (FelixAllard) [inline petclinic-frontend/src/features/carts/components/CartItem.tsx:60]: What..???? The export statement is there so i don't understand all the conflicts... Also the code seems to be working on my end...

File: petclinic-frontend/src/router.tsx
Comment (FelixAllard) [inline petclinic-frontend/src/router.tsx:1]: EXTRA IMPORT NOT NEEDED! This can cause issue... Fix this!

**Review by FelixAllard — CHANGES_REQUESTED** [review]
Please update upon my comments!

**Review by FelixAllard — DISMISSED** [review]
There is some fail in the Gateway and in Vet. Those error seems to have been imported from main and as such are not created by this commit. As such, with the correction you did, looks good to me!

**Review by NtsoaElie — DISMISSED** [review]
Everything looks fine to me.

**Review by NtsoaElie — DISMISSED** [review]

**Review by NtsoaElie — DISMISSED** [review]

**Review by NtsoaElie — COMMENTED** [review]
Changed everything we reviewed on previously and fixed the CI checks.

**Review by PaoloOabel — APPROVED** [review]
The new UI looks really good!

**Review by NtsoaElie — APPROVED** [review]
Changed everything as commented before.

**Pull request merged**